src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -498,54 +537,62 @@ private int decodeFrameType(
-                break;
-            case CLOSE:
-                frame = performative.close();
-                server.onDecodeFrameHeader(frameHeader, b -> b.close(performative.close()));
-                break;
-            default:
-                throw new IllegalStateException("Unexpected value: " + frameType);
+                server.onDecodeError(traceId, authorization, CONNECTION_FRAMING_ERROR);
+                server.decoder = decodeFrameType;
+                progress = limit;
+            }
+            else
+            {
+                final AmqpPerformativeFW performative = frameHeader.performative();
+                final AmqpDescribedType frameType = performative.kind();
+                final AmqpServerDecoder decoder = decodersByPerformative.getOrDefault(frameType, decodeUnknownType);
+                final ListFW frame;
+                switch (frameType)
+                {
+                case OPEN:
+                    frame = performative.open();
+                    server.onDecodeFrameHeader(frameHeader, b -> b.open(performative.open()));
+                    break;
+                case BEGIN:
+                    frame = performative.begin();
+                    server.onDecodeFrameHeader(frameHeader, b -> b.begin(performative.begin()));
+                    break;
+                case ATTACH:
+                    frame = performative.attach();
+                    server.onDecodeFrameHeader(frameHeader, b -> b.attach(performative.attach()));
+                    break;
+                case FLOW:
+                    frame = performative.flow();
+                    server.onDecodeFrameHeader(frameHeader, b -> b.flow(performative.flow()));
+                    break;
+                case TRANSFER:
+                    frame = performative.transfer();
+                    server.onDecodeFrameHeader(frameHeader, b -> b.transfer(performative.transfer()));
+                    break;
+                case DISPOSITION:
+                    frame = performative.disposition();
+                    server.onDecodeFrameHeader(frameHeader, b -> b.disposition(performative.disposition()));
+                    break;
+                case DETACH:
+                    frame = performative.detach();
+                    server.onDecodeFrameHeader(frameHeader, b -> b.detach(performative.detach()));
+                    break;
+                case END:
+                    frame = performative.end();
+                    server.onDecodeFrameHeader(frameHeader, b -> b.end(performative.end()));
+                    break;
+                case CLOSE:
+                    frame = performative.close();
+                    server.onDecodeFrameHeader(frameHeader, b -> b.close(performative.close()));
+                    break;
+                default:
+                    throw new IllegalStateException("Unexpected value: " + frameType);
+                }
```

Comment:
Suggest we remove this switch statement, extract whatever information we need from the frameHeader as decode member fields, and use them in `decodeOpen`, etc.